core/src/main/java/org/wso2/testgrid/core/command/RunTestPlanCommand.java

```diff
@@ -177,7 +168,13 @@ private Product getProduct(String productName)
      */
     private void deleteFile(Path filePath) throws CommandExecutionException {
         try {
-            Files.deleteIfExists(filePath);
+            if (Files.exists(filePath)) {
+                if (logger.isDebugEnabled()) {
+                    logger.debug(String.format("Deleting file: %s that has content: %s", filePath,
+                            new String(Files.readAllBytes(filePath), StandardCharsets.UTF_8)));
+                }
+                Files.deleteIfExists(filePath);
+            }
```

Comment:
We could change the same block as below.

if (Files.deleteIfExists(filePath)) {
                if (logger.isDebugEnabled()) {
                    logger.debug(String.format("Deleted file: %s that has content: %s", filePath,
                            new String(Files.readAllBytes(filePath), StandardCharsets.UTF_8)));
                }
}

Reply:
I'm printing the contents of the file before deleting. It's not available after deleting. :)

Reply:
Missed that part :)